background/src/wallet/wallet.js

```diff
@@ -63,32 +64,41 @@ export default class Wallet {
 
         return {address, encryptedWallet}
     }
-    async _transferPNG(sender, receiver, amount, nonce, memo) {
-        if (!await this.isValidNonce(nonce)) {
+    async _transferPNG(sender, receiver, amount, nonce) {
+        if (!await this.isValidNonce(sender, nonce)) {
             throw 'Invalid Nonce'
         }
 
         let senderEncryptedWallet = await this.storage.secureGet(ENCRYPTED_WALLET + sender.toLowerCase())
         let wallet = await this.decryptWallet(senderEncryptedWallet)
-        const png = {
-            decimals: 18,
-            minters: null,
+        const PNG = {
```

Comment:
Should be handled by `encodeCurrency`.

Reply:
Acutally, as far I have understood when `encodeUnsignedTxWithSystemAction()` is called, `encodeSystemAction()` is called which in turn calls `encodeTransfer()` which calls `encodeCurrency()` under the hood, so if I understood correctly I wouldn't call `encodeCurrency()` here.